osu.Game.Rulesets.Catch/Objects/Drawables/CaughtObjectContainer.cs

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Performance;
+using osu.Framework.Graphics.Pooling;
+using osu.Game.Rulesets.Catch.UI;
+using osu.Game.Utils;
+using osuTK;
+
+namespace osu.Game.Rulesets.Catch.Objects.Drawables
+{
+    public class CaughtObjectContainer : CompositeDrawable
+    {
+        public Container<DrawableCaughtObject> StackedObjectContainer { get; }
+
+        private readonly Container<DrawableCaughtObject> droppedObjectTarget;
+
+        private readonly LifetimeEntryManager lifetimeManager = new LifetimeEntryManager();
+
+        private readonly Dictionary<LifetimeEntry, DrawableCaughtObject> drawableMap = new Dictionary<LifetimeEntry, DrawableCaughtObject>();
+
+        private readonly HashSet<CaughtObjectEntry> aliveStackedObjects = new HashSet<CaughtObjectEntry>();
+        private readonly Dictionary<CaughtObjectEntry, CaughtObjectEntry> dropEntryMap = new Dictionary<CaughtObjectEntry, CaughtObjectEntry>();
+
+        private readonly DrawablePool<DrawableCaughtFruit> caughtFruitPool;
+        private readonly DrawablePool<DrawableCaughtBanana> caughtBananaPool;
+        private readonly DrawablePool<DrawableCaughtDroplet> caughtDropletPool;
+
+        /// <summary>
+        /// The randomness used to compute position in stack.
+        /// It is incremented for each <see cref="Add"/> call and decremented for each <see cref="Remove"/> call to make a replay consistent.
+        /// </summary>
+        private int randomSeed = 1;
```

Comment:
Might as well seed by index in beatmap at this point to save on tracking this.

Reply:
Because testing code doesn't set IndexInBeatmap, using it makes stacking ugly in testing.
It is fine to modify testing code, but I think tracking this value in `CaughtObjectContainer` (or maybe in `Catcher`) is better as it requires less preparation to use from outside.